app/main/lib/similarity.py

```diff
@@ -176,23 +189,28 @@ def blocking_get_similar_items(item, similarity_type):
     response = blocking_search_image(item)
     app.logger.info(f"[Alegre Similarity] [Item {item}, Similarity type: {similarity_type}] response for search was {response}")
     return response
+  elif similarity_type == "video":
+    response = video_model().blocking_search(model_response_package(item, "search"), "video")
+    app.logger.info(f"[Alegre Similarity] [Item {item}, Similarity type: {similarity_type}] response for search was {response}")
+    return response
   else:
       raise Exception(f"{similarity_type} modality not implemented for blocking requests!")
 
 def async_get_similar_items(item, similarity_type):
   app.logger.info(f"[Alegre Similarity] [Item {item}, Similarity type: {similarity_type}] searching on item")
   if similarity_type == "audio":
-    response = audio_model().async_search(model_response_package(item, "search"), "audio")
+    response, waiting_for_callback = audio_model().async_search(model_response_package(item, "search"), "audio")
     app.logger.info(f"[Alegre Similarity] [Item {item}, Similarity type: {similarity_type}] response for search was {response}")
-    return response
+    return response, waiting_for_callback
   elif similarity_type == "video":
-    response = video_model().async_search(model_response_package(item, "search"), "video")
+    response, waiting_for_callback = video_model().async_search(model_response_package(item, "search"), "video")
+    audio_response, waiting_for_audio_callback = audio_model().async_search(video_model().overload_context_to_denote_content_type(model_response_package(item, "search")), "audio")
     app.logger.info(f"[Alegre Similarity] [Item {item}, Similarity type: {similarity_type}] response for search was {response}")
-    return response
+    return response, waiting_for_callback or waiting_for_audio_callback
```

Comment:
response should be video?

Reply:
This and the audio_model() call are intentional - we need to encode the audio track for all videos to see if we can match them across modes (i.e. this MP3 matches this video's audio track, so they are able to be matched)

Reply:
Oh nice!  Can you add comments to explain?